views/default/core/friends/collection.php

```diff
@@ -1,67 +1,55 @@
 <?php
+
 /**
  * View a friends collection
  *
  * @package Elgg
  * @subpackage Core
  *
- * @uses $vars['collection'] The individual friends collection
+ * @uses $vars['collection'] The access collection
  */
-
-$coll = $vars['collection'];
-
-if (is_array($vars['collection']->members)) {
-	$count = sizeof($vars['collection']->members);
-} else {
-	$count = 0;
+$collection = elgg_extract('collection', $vars);
+if (empty($collection)) {
+	return;
 }
 
-echo "<li><h2>";
+$members = get_members_of_access_collection($collection->id) ? : array();
```

Comment:
Agreed, but then it wouldn't be a friends picker per se.
It's good UX for anything under 100 items. With collections, we should
perhaps drop the 'edit' form and use 'add members' form, and implement a
'remove member' option.
On Oct 30, 2015 6:23 PM, "Matt Beckett" <notifications@github.com> wrote:

> In views/default/core/friends/collection.php
> <https://github.com/Elgg/Elgg/pull/9085#discussion_r43528657>:
>
> >  }
> >
> > -echo "<li><h2>";
> > +$members = get_members_of_access_collection($collection->id) ? : array();
>
> If we're re-writing this we should get away from outputting *all*
> friends. These are trouble spots for scalability that we're constantly
> having to override in our projects become someone thinks it's cool to
> friend 10k people.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Elgg/Elgg/pull/9085/files#r43528657>.
>